huaweicloud/services/gaussdb/data_source_huaweicloud_gaussdb_cassandra_dedicated_resource.go

```diff
@@ -54,59 +58,66 @@ func DataSourceGeminiDBDehResource() *schema.Resource {
 	}
 }
 
-func dataSourceGeminiDBDehResourceRead(d *schema.ResourceData, meta interface{}) error {
-	config := meta.(*config.Config)
-	region := config.GetRegion(d)
-	client, err := config.GeminiDBV3Client(region)
+func dataSourceGeminiDBDehResourceRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.GeminiDBV3Client(region)
 	if err != nil {
-		return fmtp.Errorf("Error creating HuaweiCloud GaussDB client: %s", err)
+		return diag.Errorf("error creating GaussDB client: %s", err)
 	}
 
 	pages, err := instances.ListDeh(client).AllPages()
 	if err != nil {
-		return err
+		return diag.Errorf("error getting GaussDB cassandra dedicated list: %s", err)
 	}
 
 	allResources, err := instances.ExtractDehResources(pages)
 	if err != nil {
-		return fmtp.Errorf("Unable to retrieve dedicated resources: %s", err)
+		return diag.Errorf("unable to retrieve dedicated resources: %s", err)
 	}
 
-	resource_name := d.Get("resource_name").(string)
+	resourceName := d.Get("resource_name").(string)
 	refinedResources := []instances.DehResource{}
 	for _, refResource := range allResources.Resources {
 		if refResource.EngineName != "cassandra" {
 			continue
 		}
-		if resource_name != "" && refResource.ResourceName != resource_name {
+		if resourceName != "" && refResource.ResourceName != resourceName {
 			continue
 		}
 		refinedResources = append(refinedResources, refResource)
 	}
 
 	if len(refinedResources) < 1 {
-		return fmtp.Errorf("Your query returned no results. " +
-			"Please change your search criteria and try again.")
+		return diag.Errorf("your query returned no results. " +
+			"please change your search criteria and try again.")
 	}
 
 	if len(refinedResources) > 1 {
-		return fmtp.Errorf("Your query returned more than one result." +
-			" Please try a more specific search criteria")
+		return diag.Errorf("your query returned more than one result." +
+			" please try a more specific search criteria")
 	}
 
 	resource := refinedResources[0]
 
-	logp.Printf("[DEBUG] Retrieved Resource %s: %+v", resource.Id, resource)
-	d.SetId(resource.Id)
+	log.Printf("[DEBUG] Retrieved Resource %s: %+v", resource.Id, resource)
+
+	uuid, err := uuid.GenerateUUID()
```

Comment:
we can keep the resource ID as the data source only query one item.

Reply:
okay, it has been restored.

Reply:
done